src/composables/transactionList.ts

```diff
@@ -0,0 +1,329 @@
+import { useStore } from 'vuex';
+import { orderBy, uniqBy } from 'lodash-es';
+import {
+  computed,
+  onMounted,
+  ref,
+  watch,
+} from 'vue';
+import { Encoded, Tag } from '@aeternity/aepp-sdk';
+
+import type {
+  ITransaction,
+  ITransactionsState,
+  ITransactionStateList,
+} from '../types';
+import { useMiddleware } from './middleware';
+import {
+  AEX9_TRANSFER_EVENT,
+  fetchJson, getLocalStorageItem, setLocalStorageItem,
+  TX_DIRECTION,
+} from '../popup/utils';
+import { useAccounts } from './accounts';
+import JsonBig from '../lib/json-big';
+import { useAeSdk } from './aeSdk';
+
+const LOCAL_STORAGE_KEY = ['transactions_state'];
+
+const isInitalized = ref(false);
+
+const generateEmptyTransactionState = (): ITransactionsState => ({
+  loaded: [],
+  nextPageUrl: '',
+  pending: {},
+  tipWithdrawnTransactions: [],
+});
+const transactions = ref<ITransactionStateList>(
+  getLocalStorageItem(LOCAL_STORAGE_KEY) || {},
+);
+
+function getTransactionStateByAccountAddress(address: Encoded.AccountAddress) {
+  return transactions.value[address];
+}
+
+function getOrInitTransactionStateForAccount(address: Encoded.AccountAddress): ITransactionsState {
+  const accountTransactionState = getTransactionStateByAccountAddress(address);
+
+  if (accountTransactionState) {
+    return accountTransactionState;
+  }
+
+  transactions.value[address] = generateEmptyTransactionState();
+
+  return transactions.value[address];
+}
+
+function getTransactionByHash(address: Encoded.AccountAddress, hash: string) {
+  return transactions.value[address]?.loaded.find((transaction) => transaction.hash === hash);
+}
+
+function updateAccountTransactions(address: Encoded.AccountAddress, transaction: ITransaction) {
+  const transactionIndex = transactions.value[address]?.loaded?.findIndex(
+    (tr) => tr.hash === transaction.hash,
+  );
+
+  if (transactionIndex >= 0) {
+    transactions.value[address].loaded[transactionIndex] = transaction;
+  }
```

Comment:
If not already exists then maybe you can push the transaction to the array. 
This case maybe rare.